lesson16/main.js

```diff
@@ -0,0 +1,125 @@
+"use strict";
+const categoryTab = document.getElementById("js-categoryTab");
+
+const jsonURL = "http://myjson.dit.upm.es/api/bins/du3p";
+// const jsonURL = "https://myjson.dit.upm.es/api/bins/gdod";　データが空のjson
+async function fetchData(url) {
+   const response = await fetch(url);
+   const json = await response.json();
+   return json.data;
+}
+
+const createElementWithClassName = (element, name) => {
+   const createElement = document.createElement(element);
+   createElement.classList.add(name);
+   return createElement;
+};
+
+const displayMessage = (message) => {
+   const div = document.createElement("div");
+   body.appendChild(div);
+   div.textContent = message;
+}
+
+const startLoading = () => {
+   const loadingIcon = document.createElement("img");
+   loadingIcon.id = "loadingIcon";
+   loadingIcon.src = "img/loading-circle.gif";
+   body.appendChild(loadingIcon);
+};
+
+const endLoading = () => {
+   const loadingIcon = document.getElementById("loadingIcon");
+   loadingIcon.remove();
+}
+
+createTabContents();
+async function createTabContents() {
+   startLoading();
+   let responseData;
+   try {
+      responseData = await fetchData(jsonURL);
+      console.log(responseData);
+      if(responseData.length === 0) {
+         displayMessage("データはまだありません");
+         return;
+         } 
+   } catch (error) {
+      displayMessage(error);
+      console.log(error);
+   } finally {
+      endLoading();
+   }
+   createCategoryTab(responseData);
+   renderTabContainer(responseData);
+}
+
+const createCategoryTab = (values) => {
+   const fragment = document.createDocumentFragment();
+   for (let i = 0; i < values.length; i++) {
+      const categoryTabItem = createElementWithClassName("li", "category-tab__item")
+      categoryTabItem.id = `js-categoryTabItem${i}`;
+      const categoryTabText = createElementWithClassName("a", "category-tab__text");
+      categoryTabText.id = `js-categoryTabText${i}`
```

Comment:
Q.
You don't seem to be using this id, do you need it?
```
categoryTabText.id = `js-categoryTabText${i}`
```

Reply:
It's true. Thank you for pointing that out.
I removed that.
[ec8b8f8](https://github.com/taka0209/js_lesson/pull/18/commits/ec8b8f85bc532e47b37f12077105ac7e58e60ab8)[](https://github.com/taka0209)